app/custom_auth.py

```diff
@@ -0,0 +1,74 @@
+from django.contrib import auth
+from django.conf import settings
+from django.contrib.auth.models import User
+
+class CustomSSOMiddleware:
+    """
+    A custom middleware that allows SSO via HTTP header passed in the request
+    The name of HTTP header is defined in the settings.py by the variable HEADER_CUSTOM_USER_NAME  
+
+    
+    (1) If HTTP header with the user name is not passed no further processing is done by the middleware.
+    (2) If HTTP header with the user name is passed then:
+        (2.1) Authenciate the user if already exist in the database. 
+        (2.2) Create user if does not exist in the database and then autenicate the user. 
+        (2.3) User is now signed in and session cookies are set.  
+    ...
+    """
+    def __init__(self, get_response):
+        print ('LOG: Inside the custom middleware!   __init__')
+        self.get_response = get_response
+     
+    def __call__(self, request):
+        print ('LOG: Inside the custom middleware!   __call__')
+        
+        header =  settings.HEADER_CUSTOM_USER_NAME
+        username= ""
+
+        try:
+            username = request.META[header]
+        except:
```

Comment:
Scope the exception to what's necessary (here it should be `KeyError`) to avoid catching more exceptions than we expect. More specifically, always avoid `except:` since it'll also catch things like out of memory, keyboard interrupt or system errors ([more information](https://docs.python.org/2/howto/doanddont.html#except)).